tests/system/large/test_session.py

```diff
@@ -46,7 +50,72 @@
         # ),
     ],
 )
-def test_read_gbq_for_large_tables(session: Session, query_or_table, index_col):
+def test_read_gbq_for_large_tables(
+    session: bigframes.Session, query_or_table, index_col
+):
     """Verify read_gbq() is able to read large tables."""
     df = session.read_gbq(query_or_table, index_col=index_col)
     assert len(df.columns) != 0
+
+
+def test_close(session):
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    full_id_1 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    full_id_2 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    with pytest.raises(google.cloud.exceptions.Conflict):
```

Comment:
Let's explain what we're doing here:

```suggestion
    # Check that the table was actually created.
    with pytest.raises(google.cloud.exceptions.Conflict):
```

Thought: Wouldn't a call to `get_table` be a little easier to tell the intention?

Reply:
changed to `assert bqclient.get_table(full_id_1).created is not None`

Reply:
Also added a comment.